runsc/cmd/exec.go

```diff
@@ -330,7 +357,7 @@ func (ex *Exec) argsFromCLI(argv []string, enableRaw bool) (*control.ExecArgs, e
 		ExtraKGIDs:       extraKGIDs,
 		Capabilities:     caps,
 		StdioIsPty:       ex.consoleSocket != "" || console.IsPty(os.Stdin.Fd()),
-		FilePayload:      urpc.FilePayload{[]*os.File{os.Stdin, os.Stdout, os.Stderr}},
+		FilePayload:      control.NewFDMapWithOriginalFDs(os.Stdin, os.Stdout, os.Stderr),
```

Comment:
These calls will get slightly more complicated if you take my suggestion and get rid of the internal `FilePayload` type, but I don't think by much.

We could either construct the GuestFDs and FilePayload right next to eachother:

```
return &control.ExecArgs{
...
  // Guest FDs correspond 1:1 to files in FilePayload.
  GuestFDs: []int{0, 1, 2},
  FilePayload: urpc.FilePayload([]*os.File{os.Stdin, os.Stdout, os.Stderr}),
...
}
```

Or we could have a helper method on ExecArgs like `ExecArgs.AddFilesFromMap(map[int]*os.File)`

```
func (e *ExecArgs) AddFilesFromMap(m map[int]*os.File) {
  guestFDs := make([]int, 0, len(m))
  files := make([]*os.File, 0, len(m))
  for fd, file := range m {
    guestFDs = append(guestFDs, fd)
    files = append(files, file)
  }
  e.FilePayload = urpc.FilePayload(files)
  e.GuestFDs = guestFDs
}
```

Reply:
I like this!